zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroValidationService.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.service;
+
+import org.apache.shiro.realm.Realm;
+import org.apache.shiro.realm.text.IniRealm;
+import org.apache.shiro.util.ThreadContext;
+import org.apache.shiro.web.mgt.DefaultWebSecurityManager;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Collection;
+
+/**
+ * ShiroValidationService to validate shiro config
+ */
+public class ShiroValidationService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShiroValidationService.class);
+
+  @Inject
+  public ShiroValidationService(ZeppelinConfiguration conf) throws Exception {
+    LOGGER.info("ShiroValidationService is initializing");
+    init(conf);
+  }
+
+  public void init(ZeppelinConfiguration conf) throws Exception {
+    if (conf.getShiroPath().length() > 0) {
+      loadSecurityManager();
+      validateRealms();
+    }
+    LOGGER.info("ShiroValidationService is initialized.");
+  }
+
+  private synchronized void loadSecurityManager() throws Exception {
+    long waitTime = 500;
+    Integer nosOfTry = 0;
```

Comment:
I don't think it's good idea to check it with polling. How about initializing it in the `public static main` directly, checking if the configuration is correct or not and injecting it as an instance?

Reply:
Initializing it from the [public static main](url) only by Injecting as a service.

Reply:
This class is injected by `ServiceLocator` not initializing manually, What I mean is to initialize it manually and passing the instance into the injector. By doing this, you might control the SecurityManger's lifecycle.

Reply:
BTW, why is that `securityManager` null?

Reply:
Yes, we can do that but the Shiro environment (as well as SecurityManager) gets initialized when the jetty server start method gets invoked, till that time SecurityManager is null. for the same reason, I have used polling.

Reply:
AFAIK, it doesn't have `SecurityManager` to validate `shiro.ini` as it's ok to check if it's valid or not. Could you improve your logic to check the validity only?

Reply:
`SecurityManger` gets the realms from the `shiro.ini` and we validate the specified realms is valid or not. Which validity logic I need to improve?

Reply:
E.g. you could read `shiro.ini` directly and parse some information from it.

Reply:
Ok. Then there is no need to load SecurityManager and also that will help to avoid the polling approach, right? 
I am just thinking about test-cases behavior after these changes.